core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1927,4 +1927,31 @@ package object config {
       .version("3.0.1")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val PUSH_BASED_SHUFFLE_ENABLED =
+    ConfigBuilder("spark.shuffle.push.based.enabled")
+      .doc("Set to 'true' to enable push based shuffle")
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val MAX_MERGER_LOCATIONS_CACHED =
+    ConfigBuilder("spark.shuffle.push.retainedMergerLocations")
+      .doc("Max number of shuffle services hosts info cached to determine the locations of" +
```

Comment:
"shuffle services hosts info" is hard to read.  Also by reading the config I have no idea what this does so perhaps we should explain more.

Reply:
I tried describing it bit more, let me know if this looks ok or needs more info.